.github/scripts/binaryCompatibility.py

```diff
@@ -0,0 +1,78 @@
+#  Copyright Amazon.com Inc. or its affiliates.
+#  SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import json
+import os
+import xml.etree.ElementTree as ET
+
+from agithub.GitHub import GitHub
+
+
+def findall_recursive(node, element):
+    for item in node.findall(element):
+        yield item
+    for item in list(node):
+        yield from findall_recursive(item, element)
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--input', type=argparse.FileType('r'), help='Input file to parse')
+    parser.add_argument('--token', type=str, help='GitHub token')
+    parser.add_argument('--pr', type=str, help='GitHub PR')
+    parser.add_argument('--sha', type=str, help='GitHub SHA')
+    args = parser.parse_args()
+
+    file = args.input
+
+    my_body_dedupe = "Produced by binaryCompatability.py"
+    body = f"Binary incompatibility detected for commit {args.sha}.\n" \
+           f" See the uploaded artifacts from the action for details. You must bump the version number.\n\n"
+
+    incompatible = False
+    tree = ET.parse(file)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using xml.etree.ElementTree.parse to parse untrusted XML data is known to be vulnerable to XML attacks. Replace xml.etree.ElementTree.parse with its defusedxml equivalent function or make sure defusedxml.defuse_stdlib() is called https://bandit.readthedocs.io/en/latest/blacklists/blacklist_calls.html#b313-b320-xml-bad-elementtree